### PR TITLE
Add publish_tf rosparam functionality and updated launch file accordingly

### PR DESCRIPTION
**Pull request by jack-digilabs**
-adds check on rosparam publish_tf when publishing rosparam pose
-updated fiducial_slam.launch to include publish_tf and tf_publish_interval

### CODE REVIEW

**Comment by jack-digilabs** [comment]
This also addresses issue #148 . @elisabethwelburn this is the code you'll want.

**Comment by elisabethwelburn** [comment]
Hey thanks for doing this if only everyone on the internet  was as helpful as you !  :-) I found a solution that also works for the original fiducial_slam code is to set the tf_publish_interval to 0 as well :)

**Comment by jack-digilabs** [comment]
I don't think that solves the entire issue. Setting  tf_publish_interval to
0 only stops the re-publishing of the last known post transform. If
fiducial_slam detects a new fiducial it will publish the new tf. This is
line 494 or so, the call to publishTf().

**Comment by elisabethwelburn** [comment]
Thats a good point but my understanding is that  it doesn't register the new fidcials if you set it as a read only map so all fiducials are pre-defined within the global frame upon starting up the ROS node? If it's performing SLAM as well then sure it'll mess up your tfs

File: fiducial_slam/src/map.cpp [hunk 1]
```diff
@@ -195,6 +195,7 @@ Map::Map(ros::NodeHandle &nh) : tfBuffer(ros::Duration(30.0)){
     nh.param<std::string>("base_frame", baseFrame, "base_link");
 
     nh.param<float>("tf_publish_interval", tfPublishInterval, 1.0);
+    nh.param<bool>("publish_tf", publishPoseTf, false);
```
Comment (rohbotics) [inline fiducial_slam/src/map.cpp]: Please default to true.

File: fiducial_slam/src/map.cpp [hunk 2]
```diff
@@ -491,7 +492,11 @@ int Map::updatePose(vector<Observation>& obs, const ros::Time &time,
     poseTf = toMsg(outPose);
     poseTf.child_frame_id = outFrame;
     havePose = true;
-    publishTf();
+
+    if (publishPoseTf)
+    {
```
Comment (rohbotics) [inline fiducial_slam/src/map.cpp]: nit: Move the brace to the end of the previous line for if statements

File: fiducial_slam/src/map.cpp [hunk 3]
```diff
@@ -511,7 +516,7 @@ void Map::publishTf()
 void Map::update()
 {
     ros::Time now = ros::Time::now();
-    if (havePose && tfPublishInterval != 0.0 &&
+    if (publishPoseTf && havePose && tfPublishInterval != 0.0 &&
```
Comment (rohbotics) [inline fiducial_slam/src/map.cpp:518]: Is the tf publish interval check redundant here?

**Review by rohbotics — CHANGES_REQUESTED** [review]
Thanks for submitting the pull request, sorry it took so long for me to get to it.

I have some minor comments, once those are patched up I can merge this.

File: fiducial_slam/src/map.cpp (hunk 3 above)
Comment (jack-digilabs) [reply fiducial_slam/src/map.cpp:518]: Almost but not exactly. The tfPublishInterval is used when you want fiducial_slam to re-publish the TF when a fiducial isn't detected. For example, you may want TF's to be published only when a fiducial is detected so you'd set publishPoseTf=True and tfPublishInterval=0.0.

Comment (jack-digilabs) [reply fiducial_slam/src/map.cpp:518]: @rohbotics Need anything else to keep the ball rolling?

**Review by rohbotics — APPROVED** [review]
Sorry for the delay on merging this.

**Comment by rohbotics** [comment]
Thanks for your contribution, I found the same bug again and remembered this PR. Sorry for the making you wait so long.

**Pull request merged**